sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/SetFns.java

```diff
@@ -0,0 +1,528 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.transforms;
+
+import static org.apache.beam.vendor.guava.v26_0_jre.com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.beam.sdk.Pipeline;
+import org.apache.beam.sdk.transforms.join.CoGbkResult;
+import org.apache.beam.sdk.transforms.join.CoGroupByKey;
+import org.apache.beam.sdk.transforms.join.KeyedPCollectionTuple;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionList;
+import org.apache.beam.sdk.values.TupleTag;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.Iterables;
+
+public class SetFns {
```

Comment:
We should mention in the comments that we rely on the deterministic encoding of the coder similar to how we do GroupByKey.

Also, this implementation assumes that there will only be a single firing of the trigger. If there are multiple then the results are likely undefined.

Reply:
Added Comments for coders. 

Regarding triggers, I assume, multiple triggers should work if they are compatible.
It uses CGBK (internally uses flattens) , which checks if triggers compatible and windowFns are compatible. User will get results based on whatever data is triggered.

My understanding may be flawed. Please correct if you think that is not the case.

I have added a comment for compatible triggers and same windowFns.

Reply:
Your understanding is correct that users will get results based upon whatever data is fired because of a trigger but from a cursory reading of the docs, we mention doing intersect/distinct/... over PCollections and not trigger firings which could confuse a user into thinking that intersect/distinct will be over all elements in these PCollections.

Reply:
Thanks. I was thinking to add "If you have multiple triggers configured and fired, output of
   this transform will be calculated on data which is in the respective trigger."

Reply:
Thanks.Done